src/metabase/lib/field.cljc

```diff
@@ -78,6 +78,7 @@
         ;; we should look in to fixing this if we can.
         stage-columns (or (:metabase.lib.stage/cached-metadata stage)
                           (get-in stage [:lib/stage-metadata :columns])
+                          (lib.metadata.calculation/default-columns query stage-number stage)
```

Comment:
This is the actual fix. It seems potentially wasteful to calculate the default columns every time we need to resolve a column name.